src/port1.0/portconfigure.tcl

```diff
@@ -472,6 +472,20 @@ proc portconfigure::configure_get_sdkroot {sdk_version} {
         return {}
     }
 
+    # Prefer finding SDK using xcrun because it supports the DEVELOPER_DIR environment
+    # Thus CommandLinesTools SDK is returned if Xcode is unnecessary.
+    if {![catch {exec env DEVELOPER_DIR=${configure.developer_dir} xcrun --sdk macosx${sdk_version} --show-sdk-path 2> /dev/null} sdk]} {
+        return $sdk
+    }
+
+    # Return the newest-available SDK ("macosx") if it is present.
+    # This was previously the final fallback, but we want to prioritize CLT being used even if it's newer than current OS.
+    # See <https://trac.macports.org/ticket/57143>
+    if {![catch {exec env DEVELOPER_DIR=${configure.developer_dir} xcrun --sdk macosx --show-sdk-path 2> /dev/null} sdk]} {
```

Comment:
While that's the approach we did discuss, I think you should preserve the information that was documented in the comment that you removed with this change.

We should probably also keep the `ui_warn "Unable to determine location of the macOS ${sdk_version} SDK.  Using the default macOS SDK."` message.

Reply:
Added the warnings and message again